### ISSUE TITLE
add requirements-extras.txt with extra non-essential packages

### ISSUE THREAD

**Issue by londumas** [igmhub/picca#599]
add requirements-extras.txt with extra non-essential packages.
For example pk1d uses `from ROOT import TH1D, TFile, TTree, TProfile2D, TProfile`

**Comment by londumas** [comment]
@Waelthus, what are the different packages that are needed to run the Pk1d analysis that are not already in `requirements.txt`?

**Comment by londumas** [comment]
Adding this in branch https://github.com/igmhub/picca/tree/correlation_physical_model

**Comment by Waelthus** [comment]
I don't think we need any additional packages for the 1d measurement right now, that are not included.
The code contains imports of the `ROOT` python bindings, but only if you decide to use root as an output format.
As `ROOT` cannot be installed with pip/conda anyway (at least to my knowledge), I don't think it should be in `requirements.txt` (not sure if that makes it suitable for the `requirements_extra.txt`). I could put a test into the `picca_Pk1d` that checks if `ROOT` is importable and if not spits out an error message (and either breaks or sets the output to `fits`)

**Comment by londumas** [comment]
Fixed in branch https://github.com/igmhub/picca/compare/correlation_physical_model
